packages/react-native-web/src/exports/StyleSheet/modality.js

```diff
@@ -17,78 +17,47 @@
  */
 
 import { canUseDOM } from 'fbjs/lib/ExecutionEnvironment';
+import hash from '../../vendor/hash';
 
-const rule = ':focus { outline: none; }';
-let ruleExists = false;
+const modalityClassName =
+  'rn-' +
+  (process.env.NODE_ENV === 'production'
+    ? hash('rnw-modality')
+    : 'modality-' + hash('rnw-modality'));
+// These elements should always have a focus ring drawn, because they are
+// associated with switching to a keyboard modality.
+const keyboardModalityWhitelist = [
+  '[type="text"]',
+  '[type="search"]',
+  '[type="url"]',
+  '[type="tel"]',
+  '[type="email"]',
+  '[type="password"]',
+  '[type="number"]',
+  '[type="date"]',
+  '[type="month"]',
+  '[type="week"]',
+  '[type="time"]',
+  '[type="datetime"]',
+  '[type="datetime-local"]',
+  'textarea',
+  '[role="textbox"]'
+]
+  .map(s => `:not(${s})`)
+  .join('');
+
+const rule = `:focus:not(.${modalityClassName})${keyboardModalityWhitelist} { outline: none; }`;
```

Comment:
This is ugly af but it seems to perform better. It'd be great if you could test this because I changed the `keyboardModalityWhitelist` selectors a bit.